src/main/java/org/apache/mesos/hdfs/state/LiveState.java

```diff
@@ -1,103 +1,151 @@
 package org.apache.mesos.hdfs.state;
 
+import com.google.common.base.Predicate;
+import com.google.common.collect.Sets;
 import com.google.inject.Singleton;
+import org.apache.commons.lang.time.DateUtils;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.apache.mesos.Protos;
+import org.apache.mesos.hdfs.config.SchedulerConf;
 import org.apache.mesos.hdfs.util.HDFSConstants;
 
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Date;
 import java.util.HashMap;
 import java.util.HashSet;
-import java.util.Map;
+import java.util.LinkedHashMap;
 import java.util.Set;
 
 @Singleton
 public class LiveState {
-  private static final Log log = LogFactory.getLog(LiveState.class);
-  private final Map<Protos.TaskID, String> taskHostMap;
-  private final Set<Protos.TaskID> journalNodes;
-  private final Set<Protos.TaskID> nameNodes;
-  private final Map<Protos.TaskID, String> taskSlaveMap;
-  private final Set<String> journalNodeHosts;
-  private final Set<String> nameNodeHosts;
-  private final HashSet<Protos.TaskID> stagingTasks;
+  public static final Log log = LogFactory.getLog(LiveState.class);
+  private Set<Protos.TaskID> stagingTasks = new HashSet<>();
+  private AcquisitionPhase currentAcquisitionPhase = AcquisitionPhase.RECONCILING_TASKS;
+  // TODO (nicgrayson) Might need to split this out to jns, nns, and dns if dns too big
+  private LinkedHashMap<Protos.TaskID, Protos.TaskStatus> runningTasks = new LinkedHashMap<>();
+  private HashMap<Protos.TaskStatus, Boolean> nameNode1TaskMap = new HashMap<>();
+  private HashMap<Protos.TaskStatus, Boolean> nameNode2TaskMap = new HashMap<>();
+  private Timestamp ReconciliationTimestamp;
+  private SchedulerConf schedulerConf = null;
+    
+  public LiveState(SchedulerConf conf) {
+    schedulerConf = conf;
+  }
 
-  public LiveState() {
-    taskHostMap = new HashMap<>();
-    taskSlaveMap = new HashMap<>();
-    journalNodes = new HashSet<>();
-    nameNodes = new HashSet<>();
-    journalNodeHosts = new HashSet<>();
-    nameNodeHosts = new HashSet<>();
-    stagingTasks = new HashSet<>();
+  public boolean reconciliationComplete() {
+    return ReconciliationTimestamp.before(new Date());
   }
 
-  public Map<Protos.TaskID, String> getTaskHostMap() {
-    return taskHostMap;
+  public boolean isNameNode1Initialized() {
+    return !nameNode1TaskMap.isEmpty() && nameNode1TaskMap.values().iterator().next();
```

Comment:
Maybe these should check that the TaskStatus.message is INIT/BOOTSTRAP, or use TASK_STARTING/RUNNING to differentiate between started/allocated and initialized